msu/hooks/items/weapons/weapon.nut

```diff
@@ -92,6 +88,7 @@
 
 	q.addWeaponType <- function( _weaponType, _setupCategories = true )
 	{
+		this.setupWeaponType(false);
```

Comment:
I don't think this works. Example:
- A vanilla weapon that does `this.m.Categories = "Axe"` in its `create()`.
- Mod A hooks this function, calls the `__original()` and does `addWeaponType(Spear)`.
- This goes to `setupWeaponType(false)` which sets the WeaponType to `None`. Then goes through all the categories, sets it to `Axe`. Then the `addWeaponType` function set it to `Axe | Spear`. Note that the Categories are not updated/set here.
- Mod B hooks afterward and says `addWeaponType(Hammer)`.
- This goes to `setupWeaponType(false)` which sets the WeaponType to `None`. Then goes through all the categories, sets it to `Axe`. Then the `addWeaponType` function set it to `Axe | Hammer`. Note that this means that the `Spear` added by Mod A is lost.

I guess the solution is to also set up the categories immediately after `setupWeaponType` in `addWeaponType` and `removeWeaponType`.